activerecord/lib/active_record/persistence.rb

```diff
@@ -460,14 +460,19 @@ def reload(options = nil)
     #   ball = Ball.new
     #   ball.touch(:updated_at)   # => raises ActiveRecordError
     #
-    def touch(*names)
+    def touch(*args)
       raise ActiveRecordError, "cannot touch on a new record object" unless persisted?
 
       attributes = timestamp_attributes_for_update_in_model
+      time_opt, names = args.partition{|arg| arg.kind_of?(Hash) && arg.has_key?(:time)}
       attributes.concat(names)
 
       unless attributes.empty?
-        current_time = current_time_from_proper_timezone
+        unless time_opt.empty?
+            current_time = time_opt[0][:time]
+        else
+            current_time = current_time_from_proper_timezone
+        end
```

Comment:
It looks like your tab indentation is set to 4 spaces and it should be 2 spaces. Let me know what your text editor is if you can't figure out how to change your indentation.